### PR TITLE
change default feature vector format (to 'dense' from 'sparse')

### PR DESCRIPTION
**Pull request by cpoerschke**
also: increase test coverage w.r.t. 'sparse' vs. 'dense' vs. 'default' feature vector format

(includes #200 changes)

### CODE REVIEW

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/response/transform/LTRFeatureLoggerTransformerFactory.java [hunk 1]
```diff
@@ -77,7 +77,7 @@
   private String fvCacheName;
   private String loggingModelName = DEFAULT_LOGGING_MODEL_NAME;
   private String defaultStore;
-  private String defaultFormat;
+  private FeatureLogger.FeatureFormat defaultFormat = FeatureLogger.FeatureFormat.DENSE;
```
Comment (diegoceccarelli) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/response/transform/LTRFeatureLoggerTransformerFactory.java]: `final` ?

Comment (diegoceccarelli) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/response/transform/LTRFeatureLoggerTransformerFactory.java]: maybe also `static` ?

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/response/transform/LTRFeatureLoggerTransformerFactory.java [hunk 2]
```diff
@@ -147,23 +150,30 @@ public DocTransformer create(String name, SolrParams localparams,
...
     }
     if (fvCacheName == null) {
       throw new IllegalArgumentException("a fvCacheName must be configured");
     }
-    return new CSVFeatureLogger(fvCacheName, f, csvKeyValueDelimiter, csvFeatureSeparator);
+    return new CSVFeatureLogger(fvCacheName, format, csvKeyValueDelimiter, csvFeatureSeparator);
+  }
+
+  private static FeatureLogger.FeatureFormat getFeatureFormat(String featureFormat) {
+    if (featureFormat.equals("sparse")) {
```
Comment (diegoceccarelli) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/response/transform/LTRFeatureLoggerTransformerFactory.java]: minor: `FeatureLogger.FeatureFormat.valueOf(String) ? ` also, valueOf throws `IllegalArgumentException` if the string is not an enum value, maybe it is better than returning `null`?

File: solr/contrib/ltr/src/test-files/solr/collection1/conf/solrconfig-ltr.xml [hunk 3]
```diff
@@ -36,6 +36,7 @@
   enclosed between brackets (in this case [fv]). In order to get the feature
   vector you will have to specify that you want the field (e.g., fl="*,[fv]) -->
  <transformer name="fv" class="org.apache.solr.ltr.response.transform.LTRFeatureLoggerTransformerFactory">
+   <str name="defaultFormat">${solr.ltr.transformer.fv.defaultFormat:dense}</str>
```
Comment (diegoceccarelli) [inline solr/contrib/ltr/src/test-files/solr/collection1/conf/solrconfig-ltr.xml]: nice

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/response/transform/LTRFeatureLoggerTransformerFactory.java (hunk 1 above)
Comment (cpoerschke) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/response/transform/LTRFeatureLoggerTransformerFactory.java]: There is a setter (`setDefaultFormat`), the intention behind that setter is that the `solrconfig.xml` can configure a default and then the `fl=[features]` clause need not specify a `format` (unless it wishes to).

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/response/transform/LTRFeatureLoggerTransformerFactory.java (hunk 2 above)
Comment (cpoerschke) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/response/transform/LTRFeatureLoggerTransformerFactory.java]: thanks for that! incorporated in the most recent commit.

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/response/transform/LTRFeatureLoggerTransformerFactory.java (hunk 1 above)
Comment (diegoceccarelli) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/response/transform/LTRFeatureLoggerTransformerFactory.java]: ok

**Pull request merged**